library/ssl_tls.c

```diff
@@ -5227,16 +5238,17 @@ static int ssl_check_dtls_clihlo_cookie(
  * that looks like a ClientHello.
  *
  * - if the input looks like a ClientHello without cookies,
- *   send back HelloVerifyRequest, then
- *   return MBEDTLS_ERR_SSL_HELLO_VERIFY_REQUIRED
+ *   send back HelloVerifyRequest, then return 0
```

Comment:
"send back HelloVerifyRequest" Was the behavior of previous 2.16.4/5 versions ?

Reply:
The behavior was made different for the baremetal branch, and a fix was submitted for 2.16 and then ported in https://github.com/ARMmbed/mbedtls/pull/3149 to baremetal. It is now incorporated in this PR.